server/events/eventController.js

```diff
@@ -22,7 +22,30 @@ module.exports = {
 
   //To do: Accept an object argument and filter based on it
   getEvents: function(req, res, next) {
-    //if accept an object argument need to somehow grab it? or let frontend do it?
+    //If looking for a specific event
+    if(request.query.eventId !== undefined) {
+      Event.findById(request.query.eventId).exec(function (err, events) {
+        if(err) { return console.error(err); }
+        res.json(events);
+      })
+    }
+
+    //If looking for a limited number of events, returns the soonest to occur
+    if(request.query.limit !== undefined) {
+      Event.find().exec(function (err, events) {
+        if(err) { return console.error(err); }
+        
+        var returnVal;
+        returnVal = events.sort(function(a, b) {
+            return b.start_time - a.start_time;
+          })
+          .slice(0, request.query.limit)
+
+        res.json(returnVal);
+      })
+    }
+
+    //If not queries are included, return all events
     Event.find().exec(function (err, events) {
```

Comment:
there was no return within the getEvent function, so line 49 execute no matter what. 
When the query was executed and express is ready to send the response back, it throw me an error saying the the response had already been send (by either of the if block in the lines before)